flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -407,15 +404,9 @@ private void startTaskExecutorInContainer(Container container) {
 				containerIdStr,
 				container.getNodeId().getHost());
 
-			nodeManagerClient.startContainer(container, taskExecutorLaunchContext);
+			nodeManagerClient.startContainerAsync(container, taskExecutorLaunchContext);
 		} catch (Throwable t) {
-			log.error("Could not start TaskManager in container {}.", container.getId(), t);
-
-			// release the failed container
-			workerNodeMap.remove(resourceId);
-			resourceManagerClient.releaseAssignedContainer(container.getId());
-			// and ask for a new one
-			requestYarnContainerIfRequired();
+			onStartContainerError(container.getId(), t);
```

Comment:
The same here. How can we react to start container failure? I don't think it handles the future result of an async method.